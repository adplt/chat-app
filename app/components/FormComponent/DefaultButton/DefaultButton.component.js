import React from 'react';
import PropTypes from 'prop-types';
import {View, Text} from 'react-native';
import styles from './DefaultButton.styles';
import Touchable from '../../TouchableComponent/Touchable.component';

const DefaultButton = (props) => {
  const {
    children,
    disabled = false,
    onPress,
    highlightColor = 'white',
    highlightOpacity = 0.5,
    text,
    style = {},
    textStyle = {},
    disabledStyle = {},
    theme = 'primary'
  } = props;
  let viewStyle = [theme === 'primary' ? styles.buttonPrimary : styles.buttonSecondary, style];
  let buttonTextStyle = [theme === 'primary' ? styles.buttonTextPrimary : styles.buttonTextSecondary, textStyle];
  if (disabled) {
    viewStyle = [style, styles.buttonDisabled, disabledStyle];
    buttonTextStyle = [styles.buttonText, styles.buttonTextDisabled];
  }
  const nestedChildren = children || (<Text style={buttonTextStyle}>{text}</Text>);
  return (
    <Touchable onPress={onPress} disabled={disabled} highlightColor={highlightColor} activeOpacity={highlightOpacity}>
      <View style={viewStyle}>
        {nestedChildren}
      </View>
    </Touchable>
  );
};

DefaultButton.propTypes = {
  disabled: PropTypes.bool,
  onPress: PropTypes.func,
  children: PropTypes.node,
  highlightOpacity: PropTypes.number,
  disabledStyle: PropTypes.object,
  textStyle: PropTypes.object,
  text: PropTypes.string,
  buttonType: PropTypes.string,
  style: PropTypes.object,
  highlightColor: PropTypes.string,
  theme: PropTypes.string,
};

export default DefaultButton;
