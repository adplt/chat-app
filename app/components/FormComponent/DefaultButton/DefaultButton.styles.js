import {theme} from '../../../styles/core.style';
import {
  containerRowCenter,
  fontSizeNormal,
  textLight,
} from '../../../styles/common.style';

const button = {
  ...containerRowCenter,
  borderRadius: 2,
  paddingHorizontal: 15,
  height: 56,
};

const buttonText = {
  ...fontSizeNormal,
  ...textLight,
  fontWeight: theme.fontWeightBold,
};

export default {
  container: {
    ...containerRowCenter,
    backgroundColor: theme.buttonBg,
    borderRadius: 30,
    paddingHorizontal: 15,
    height: 56
  },
  buttonPrimary: {
    ...button,
    backgroundColor: theme.primary,
  },
  buttonSecondary: {
    ...button,
    backgroundColor: theme.secondary,
  },
  buttonDisabled: {
    backgroundColor: theme.buttonDisabledBg,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    height: 56
  },
  buttonTextPrimary: {
    ...buttonText,
    color: theme.secondary,
  },
  buttonTextSecondary: {
    ...buttonText,
    color: theme.primary,
  },
  buttonTextDisabled: {
    ...fontSizeNormal,
    ...textLight,
    fontWeight: theme.fontWeightBold,
  },
};
