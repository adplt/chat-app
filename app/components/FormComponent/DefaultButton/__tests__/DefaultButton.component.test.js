import React from 'react';
import renderer from 'react-test-renderer';
import DefaultButton from '../DefaultButton.component';

describe('Default Button component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DefaultButton/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
