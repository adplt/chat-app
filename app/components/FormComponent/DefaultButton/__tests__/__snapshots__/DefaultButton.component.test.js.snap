// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Button component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#392d2c",
          "borderRadius": 2,
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "center",
          "paddingHorizontal": 15,
        },
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#F5F5F5",
            "fontSize": 17,
            "fontWeight": "800",
          },
          Object {},
        ]
      }
    />
  </View>
</View>
`;
